contentScript.js

```diff
@@ -1,33 +1,44 @@
-    const altText = image.getAttribute('alt');
-    if (!altText) {
-        image.classList.add('github-a11y-img-missing-alt')
-    } else {
-        const closestParagraph = image.closest('p');
-        if (!closestParagraph) return; // TODO: handle when image is nested in elements like a table cell.
 
-        closestParagraph.classList.add('github-a11y-img-container');
-
-        const subtitle = document.createElement('span');
-        subtitle.setAttribute('aria-hidden', 'true');
-        subtitle.textContent = altText;
-        subtitle.classList.add('github-a11y-img-caption');
-        
-        image.insertAdjacentElement('afterend', subtitle);
-    }
+ function appendAccessibilityInfo() {
+  document.querySelectorAll('.markdown-body').forEach(function(commentBody) {
+    // Adds alt image overlay. This is hidden from accesibility tree.
+    commentBody.querySelectorAll('img').forEach(function(image) {
+      const altText = image.getAttribute('alt');
+      if (!altText) {
+          image.classList.add('github-a11y-img-missing-alt')
+      } else {
+          const closestParagraph = image.closest('p');
+          if (!closestParagraph) return; // TODO: handle when image is nested in elements like a table cell.
+  
+          closestParagraph.classList.add('github-a11y-img-container');
+  
+          const subtitle = document.createElement('span');
+          subtitle.setAttribute('aria-hidden', 'true');
+          subtitle.textContent = altText;
+          subtitle.classList.add('github-a11y-img-caption');
+          
+          image.insertAdjacentElement('afterend', subtitle);
+      }
+    });
+  
+    // Appends heading level to headings. This is hidden from accesibility tree.
+    commentBody.querySelectorAll('h1, h2, h3, h4, h5, h6').forEach(function(heading) {
+      heading.classList.add('github-a11y-heading');
+      const headingPrefix = document.createElement('span');
+  
+      headingPrefix.setAttribute('aria-hidden', 'true');
+      headingPrefix.classList.add('github-a11y-heading-prefix');
+      headingPrefix.textContent = ` ${heading.tagName.toLowerCase()}`;
+  
+      heading.append(headingPrefix);
+    });
   });
+ }
 
-  // Appends heading level to headings. This is hidden from accesibility tree.
-  commentBody.querySelectorAll('h1, h2, h3, h4, h5, h6').forEach(function(heading) {
-    heading.classList.add('github-a11y-heading');
-    const headingPrefix = document.createElement('span');
-
-    headingPrefix.setAttribute('aria-hidden', 'true');
-    headingPrefix.classList.add('github-a11y-heading-prefix');
-    headingPrefix.textContent = ` ${heading.tagName.toLowerCase()}`;
-
-    heading.append(headingPrefix);
-  });
+ chrome.runtime.onMessage.addListener(async (message) => {
+  document.querySelectorAll(".github-a11y-img-caption").forEach(el => el.remove());
+  document.querySelectorAll(".github-a11y-heading-prefix").forEach(el => el.remove());
+  appendAccessibilityInfo();
 });
+
+appendAccessibilityInfo();
```

Comment:
This method will run on initial page load after DOM is complete.




[From Content Script Chrome Docs: ](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#run_time)

> Content scripts running at "document_idle" do not need to listen for the window.onload event, they are guaranteed to run after the DOM is complete. If a script definitely needs to run after window.onload, the extension can check if onload has already fired by using the [document.readyState](https://www.whatwg.org/specs/web-apps/current-work/#dom-document-readystate) property.